flink-runtime/src/test/java/org/apache/flink/runtime/deployment/ResultPartitionDeploymentDescriptorTest.java

```diff
@@ -104,6 +104,24 @@ public void testSerializationWithNettyShuffleDescriptor() throws Exception {
 		assertThat(shuffleDescriptorCopy.getConnectionId(), is(connectionID));
 	}
 
+	@Test
+	public void testManagedExternallyFlag() {
+		for (ResultPartitionType partitionType : ResultPartitionType.values()) {
+			ResultPartitionDeploymentDescriptor partitionDescriptor = new ResultPartitionDeploymentDescriptor(
+				new PartitionDescriptor(resultId, partitionId, partitionType, numberOfSubpartitions, connectionIndex),
+				(ShuffleDescriptor) ResultPartitionID::new,
```

Comment:
It seems explicit type cast not needed here.

Reply:
addressed in #8654.